### DISCUSSION TITLE
GitHub cask not ready to go on install? 'glibconfig.h' file not found

### DISCUSSION THREAD

**Discussion by maurymarkowitz** [Homebrew/discussions#3298] [Casks]
I have a small project that uses glib-2.0 for list and tree support. I built this on Xcode on Intel and now I'm trying to port it to my M1. I found that brew moved the install directory (good!) to /opt/homebrew (excellent) and sure enough if I go here:

/opt/homebrew/Cellar/glib/2.72.1/include/glib-2.0

I can see glib.h and all the rest I would expect. However, when I add that path to my project and build, I get:

/opt/homebrew/include/glib-2.0/glib/gtypes.h:32:10: 'glibconfig.h' file not found

Note the path - it is finding the glib install OK.

I did a find, and sure enough glibconfig.h doesn't exist. Google seems to suggest I need to run pkg_config, but that doesn't exist on the Mac (??). I would have also thought that any post-install steps like this would have been performed by brew.

Does anyone know the secret sauce here?

**Comment by SMillerDev** [comment]
>  I need to run pkg_config, but that doesn't exist on the Mac 

Neither does glib, until you install it.

**Comment by carlocab** [comment]
Try
```
brew install pkg-config
```
I also suggest avoiding Cellar paths entirely, as these can change without any warning. Instead, the path you want to include is
```
/opt/homebrew/opt/glib/include/glib-2.0
```
Using
```
/opt/homebrew/include/glib-2.0
```
will probably work too.

**Reply by maurymarkowitz** [reply]
I used the later and that definitely is finding glib.h, which is all that I import in my project. I removed the Cellars.

I was under the mistaken impression that pkg-config was required, but I see now all it does is output the directory paths for your gcc build line. As we're specifying that directly, pkg-config is not needed.

The issue is ultimately that the .h file in question is inside the /lib for some reason, a directory tree that is otherwise empty. In any event, some find poking found it and now I can compile at last.
